### ISSUE TITLE
[BUG] Plotting Stuck on 1.2.3 was working on 1.2.0 - 73%-75%

### ISSUE THREAD

**Issue by x3style** [Chia-Network/chia-blockchain#7643]
**Describe the bug**
Plotting runs normally until around 73%-75% then it gets stuck at one of the uniform sorts.
Process is online and uses proper CPU 8% but memory usage is very very low 800KB and does not increase as if its blocked in a loop.
Disk activity stops completely nothing is read and nothing is written. 

**To Reproduce**

I make 3 plots simultaneously.
2 of them finish without a problem
When the 2 finish the third get stuck i think or maybe it's stuck beforehand i can't tell for sure.
They don't get stuck on the same drive either.
Not sure if it's always the last plot i trigger, because you start the 3 plots sequentially.

**Expected behavior**
I expect all 3 to finish.

**Screenshots**
![image](https://user-images.githubusercontent.com/84071923/126223810-1b27d7f5-dcef-494d-aa15-526cc793538d.png)
![image](https://user-images.githubusercontent.com/84071923/126223840-1c644c47-0b7b-43f3-8d61-435f7ea6f80d.png)
![image](https://user-images.githubusercontent.com/84071923/126224081-79ec1bee-2f60-4247-afa6-f305691c1122.png)
![image](https://user-images.githubusercontent.com/84071923/126224227-c14cbd5b-bf75-429b-85ad-3e21917679ef.png)

**Desktop**

- OS: Windows 10 Pro
- OS Version/Flavor: 21H1 19043.1081
- CPU: I7 5820K

**Additional context**
I noticed it happening after i updated from 1.2.0 to 1.2.2

**Comment by x3style** [comment]
Happened again on a different drive same uniform sort, different drive, different %

![image](https://user-images.githubusercontent.com/84071923/126368398-318c2ca2-8fba-4639-a293-b40129666db3.png)
![image](https://user-images.githubusercontent.com/84071923/126368505-943f5580-e5f7-4102-a604-fa2612f2244f.png)

**Comment by x3style** [comment]
I ended up removing the drives from the system. I tried full formatting, deep scans, surface scans, HDD diagnose. All green and drives work fine for anything else except plotting. 

New drive works fine for plotting, but it begs the question, why only plotting is affected and all other uses are fine.

Something does not add up.

**Comment by x3style** [comment]
Issue persists but is sporadic, it comes and goes, seems to not be related to one particular HDD.

**Comment by x3style** [comment]
And keeps on going randomly on any of the HDD.
![image](https://user-images.githubusercontent.com/84071923/129486300-0f773374-85e1-4ce9-928f-d3586f9c086d.png)

**Comment by Jcragons** [comment]
i confirm it the same here on debian, since the update, i was something like 1.1. and know on the last one (1.2.3) it's almost every time. it's not using gui, only cli. generally stop at phase 3/4, almost the same than screenshot here. i've update for ploting for pool, but difficult with that ...

**Comment by x3style** [comment]
I've noticed that every time after this happens if you try to close Chia it hangs and the chia processes the farmer and the wallet are stuck and need to be force shut down, so something happens deeper into the software that makes this hang. If you reboot it works for a while then it breaks again after 48/72 hours maybe.

**Comment by Jcragons** [comment]
i've tried a few times to reboot, it was doing that directly, no 48h of plotting joy. i've just try to change k to 33 compare to 32 for testing plot size and now it have work, so maybe it's random, or link to a size (storage or memory) or just the position of the moon ;) but definitely annoying.

**Comment by x3style** [comment]
So, I've suspected it might be the memory however I ran many memory tests including the memtest86 for many hours and none ever suggested a memory issue, but ever so often my pc has a crash or reboot every couple of days which I attributed to other things.

I've since changed the RAM even though there was no technical or provable indication that the RAM was faulty, and so far in the last few days the performance is rock solid no more crashes and the plotting works. I'll come back after more time passes.

But it looks like the Plotting process is extremely sensible to RAM issues more so than the RAM testing programs, a small corruption would cause issues in the plots however small.
